src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -2178,9 +2179,93 @@ static public TimeZoneInfo FindSystemTimeZoneById(string id)
             return result;
         }
 
-        static public ReadOnlyCollection<TimeZoneInfo> GetSystemTimeZones() {
-            // UNIXTODO
-            throw new NotImplementedException();
+        /// <summary>
+        /// Returns a collection of TimeZoneInfo instances containing all valid TimeZones
+        /// from the local machine.  The entries in the collection are sorted by
+        /// 'DisplayName'.
+        ///
+        /// This method does *not* throw TimeZoneNotFoundException or
+        /// InvalidTimeZoneException, but it can throw a SecurityException.
+        /// </summary>
+        static public ReadOnlyCollection<TimeZoneInfo> GetSystemTimeZones()
+        {
+            // UNIXTODO: Issue #2497 use CachedData
+
+            List<TimeZoneInfo> timeZoneInfos = new List<TimeZoneInfo>();
+
+            string timeZoneDirectory = GetTimeZoneDirectory();
+            foreach (string timeZoneId in GetTimeZoneIds(timeZoneDirectory))
+            {
+                if (!string.IsNullOrEmpty(timeZoneId))
+                {
+                    try
+                    {
+                        TimeZoneInfo timeZoneInfo = FindSystemTimeZoneById(timeZoneId);
+                        timeZoneInfos.Add(timeZoneInfo);
+                    }
+                    catch (InvalidTimeZoneException) { }
+                    catch (TimeZoneNotFoundException) { }
+                }
+            }
+
+            // sort and copy the TimeZoneInfo's into a ReadOnlyCollection for the user
+            timeZoneInfos.Sort(new TimeZoneInfoComparer());
```

Comment:
we have no plan to expose  ClearCachedData in the .Net core because it is not right to depend on that. also calling the easter egg will require the app to know the system has some TZ change which is a challenging issue for some apps. we can keep what you are doing for now (I mean the caching) and then we can change this later if we need to.